core/modules/qdisp/QdispPool.cc

```diff
@@ -191,4 +194,37 @@ std::string PriorityQueue::statsStr() {
     return os.str();
 }
 
+
+void QdispPool::_setup(bool unitTest) {
+    if (not unitTest) {
+        // Numbers are based on 1200 threads in the _pool. Large results
+        // tend to be slow to give up their threads, thus can't be allowed
+        // to eat up the pool. Bandwidth also makes running many of the
+        // slow queries at the same time a burden on the system.
+        // TODO: Set up thread pool size and queues in configuration. DM-10237
+        _prQueue = std::make_shared<PriorityQueue>(100, 1, 5); // default (lowest) priority.
+        _pool = util::ThreadPool::newThreadPool(1200, _prQueue);
```

Comment:
I think the number should be derived based on a value reported by:
https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency

In this case you won't need to have two separate branches, for "production" and for "unit testing"

Reply:
You can guess the number of cores (in any case it will be much lower than 1200) but it does not mean you can use all of them.

Reply:
One more thing that I don't quite understand - it looks like czar can run OK on my machine (and it does show ~1200 threads) but unit test fails. Is it because unit test makes multiple thread pools?

Reply:
The core (or hardware thread) count is one limit. Another one comes from both the system level on a number of processes (threads are counted as "processes" in Linux) as well as a per-process limit. The later could be tested (for a given PID=18254) by:
    cat /proc/18254/limits 
    Limit                     Soft Limit           Hard Limit           Units     
    .. 
    Max processes             4096                 43046                processes 
    ..

I got this number on my VM which has 5 cores. However, the unit test (its older version) still fails here with the number of threads of 1200. Obviously, someone else within the same process opens a few more thousand of threads.

Reply:
And the number of system-wide threads can be seen by:
% cat /proc/sys/kernel/threads-max
86093

Reply:
I've installed "atop" on my VM (where I have the above mentioned user-level process limit of 4096) and tested the older version of the unit test. I could quickly see that the test reaches above 4000 threads when it runs. "atop" is a good application which allows to see this.

Reply:
Outside of unit tests, it should only need to create one pool. The pools do cleanup after themselves, so I'm not sure why we're getting significantly more than 1200 threads. The unit tests really only need a few threads in the pool. The extra threads may be coming from XrdSsiService (base of XrdSsiServiceMock)